tests/blockchain/eth/contracts/main/staking_escrow/test_sampling.py

```diff
@@ -43,16 +42,18 @@ def test_sampling(testerchain, token, escrow_contract):
         testerchain.wait_for_receipt(tx)
         amount = amount // 2
 
-    # Cant't use sample without points or with zero periods value
-    with pytest.raises((TransactionFailed, ValueError)):
+    # Can't use sample without points, with zero periods value, or not sorted in ascending order.
+    with pytest.raises(TransactionFailed):
         escrow.functions.sample([], 1).call()
-    with pytest.raises((TransactionFailed, ValueError)):
```

Comment:
Just a note - By removing the handling of `ValueError`, the test suite is less compatible with other backends, like geth, since the exception is EthereumTester-specific.   Unsure if this is a reasonable goal for us, but I'd like to remain open to the possibility of running the test suite without eth-tester.

Reply:
Oh so that's why! I always found weird those lines. In any case, if that's the reason, isn't it better that we have something like:

```python
TransactionException = TransactionFailed if using_eth_tester() else ValueError
...
with pytest.raises(TransactionException):
    foo.bar()
```

Otherwise, there may be the case that in our current setting the method being tested is currently raising a legitimate`ValueError` not caused by a failed transaction and we're accepting that as valid.